analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -30,6 +32,32 @@ public class NullPointerDereference : NullPointerDereferenceBase
 
         protected override DiagnosticDescriptor Rule => S2259;
 
-        public override bool ShouldExecute() => true;    // ToDo: Implement rule-specific logic
+        public override bool ShouldExecute()
+        {
+            var walker = new SyntaxKindWalker();
+            walker.SafeVisit(Node);
+            return walker.Result;
```

Comment:
I wrote `return walker.SafeVisit(Node) && walker.Result` for the review, but reverted it.
It will not bring value and will cause coverage problem. `SafeVisit=False` implies also `Result=false` and we very rarely test the result of SafeVisit in the codebase